### PR TITLE
feat: add support to style calendar cells based on cell date

### PR DESCRIPTION
**Pull request by TechSaq**
Hi **@acro5piano**

As per discussion, I did changes to ad the support to provide styles to calendar cells based on the date of the cell. For this, I have to remove evenCellBg and oddCellBg.

I introduces 2 new props `calendarCellStyle` and `calendarCellTextStyle`.

```
`calendarCellStyle`: Use this prop to provide custom styles to calendar cells based on date of cell.
calendarCellStyle = ViewStyle | (date: dayjs.Dayjs, hourRowIndex?: Number | undefined) => ViewStyle)
```

```
`calendarCellTextStyle` : Use this prop to provide custom styles to calendar cells text for Month mode. It is needed if have text color of cell visible based on date
calendarCellTextStyle = ViewStyle | (date: dayjs.Dayjs, hourRowIndex?: Number | undefined) => ViewStyle)
```

Note: `hourRowIndex` can be used to give background color to even cell and odd cells.

I have added 4 stories in mobile mode to demo the same.
1. Month - Calendar Cell Style
2. Month - Event Cell - Even and Odd Row BgColor
3. Week - Calendar Cell Style
4. Week - Calendar Cell - Even and Odd Row BgColor

**@acro5piano** Please review the changes and let me know of any further changes.

PS: `yarn build` and `yarn test` both run successfully.

### CODE REVIEW

File: src/components/HourGuideCell.tsx [hunk 1]
```diff
@@ -28,7 +39,7 @@ export const HourGuideCell = ({ cellHeight, onPress, date, hour, index }: HourGu
           u['border-b'],
           { borderColor: theme.palette.gray['200'] },
           { height: cellHeight },
-          { backgroundColor: isPair(index) ? evenCellBg : oddCellBg },
```
Comment (acro5piano) [inline src/components/HourGuideCell.tsx:31]: I think `isPair` function can be deleted safely (from util.ts)

File: src/interfaces.ts [hunk 2]
```diff
@@ -28,6 +29,10 @@ export type Mode = '3days' | 'week' | 'day' | 'custom' | 'month'
 
 export type EventCellStyle<T extends ICalendarEventBase> = ViewStyle | ((event: T) => ViewStyle)
 
+export type CalendarCellStyle =
+  | ViewStyle
+  | ((date: dayjs.Dayjs, hourRowIndex?: Number) => ViewStyle)
```
Comment (acro5piano) [inline src/interfaces.ts:34]: The function style prop `CalendarCellStyle` should accept `Date` type as consistency. and `hourRowIndex` should not accept undefined

```suggestion
export type CalendarCellStyle =
  | ViewStyle
  | ((date: Date, hourRowIndex: number) => ViewStyle)
```

**Review by acro5piano — CHANGES_REQUESTED** [review]
Sorry for my slow response and thank you for the wonderful code! This architecture is better than the current one ;) Please fix a small issue.

File: src/components/HourGuideCell.tsx (hunk 1 above)
Comment (iranaahsanali) [reply src/components/HourGuideCell.tsx:31]: @TechSaq I am not able to link this project locally for adding / fixing an issue, can you share how you did it ?

Comment (TechSaq) [reply src/components/HourGuideCell.tsx:31]: @iranaahsanali Can you please share what issue are you getting? Please share your system details also. I also got a issue due to different Node version for my Windows 10 system. If you are also having the same issue, I will be happy to help.

Comment (iranaahsanali) [reply src/components/HourGuideCell.tsx:31]: Hi, @TechSaq @acro5piano   Actually my requirements were a little bit different from this repo, so I forked this repo, and made some changes on it, I was testing the changes through storybook build, So after making changes I pushed that code to my repo. So when I installed the package in my project from my forked repo, it is throwing an error, main module was not resolved. Please check the screenshot.

Simulator Screen Shot - iPad (9th generation) - 2022-01-15 at 22 06 01

**Comment by acro5piano** [comment]
@TechSaq Can you fix my reviews? Thanks

File: src/components/HourGuideCell.tsx (hunk 1 above)
Comment (acro5piano) [reply src/components/HourGuideCell.tsx:31]: @iranaahsanali That is off-topic. Please create a new issue instead. Thanks

**Comment by TechSaq** [comment]
> @TechSaq Can you fix my reviews? Thanks

Hi @acro5piano, Yes sure I will apply your changes as suggested for sure. I am a little busy with some other stuff. I will start on it within 2-3 days.

**Comment by acro5piano** [comment]
@TechSaq Thanks!

**Comment by TechSaq** [comment]
@acro5piano Please review my changes again. I did remove isPair and applied some typescript fixes.

**Review by acro5piano — APPROVED** [review]

**Comment by acro5piano** [comment]
@TechSaq Thanks for your contribution!

**Comment by acro5piano** [comment]
Released on v3.0.0

**Pull request merged**